src/TrezorKeyring.ts

```diff
@@ -281,44 +341,48 @@ class TrezorKeyring extends EventEmitter {
 
   /**
    *
-   * @template {TypedTransaction | OldEthJsTransaction} Transaction
-   * @param {string} address - Hex string address.
-   * @param {number} chainId - Chain ID
-   * @param {Transaction} tx - Instance of either new-style or old-style ethereumjs transaction.
-   * @param {(import('trezor-connect').EthereumSignedTx) => Transaction} handleSigning - Converts signed transaction
+   * @param address - Hex string address.
+   * @param chainId - Chain ID
+   * @param tx - Instance of either new-style or old-style ethereumjs transaction.
+   * @param handleSigning - Converts signed transaction
    * to the same new-style or old-style ethereumjs-tx.
-   * @returns {Promise<Transaction>} The signed transaction, an instance of either new-style or old-style
+   * @returns The signed transaction, an instance of either new-style or old-style
    * ethereumjs transaction.
    */
-  async _signTransaction(address, chainId, tx, handleSigning) {
-    let transaction;
+  async #signTransaction<T extends TypedTransaction | OldEthJsTransaction>(
+    address: string,
+    chainId: number,
+    tx: T,
+    handleSigning: (tx: EthereumSignedTx) => T,
+  ): Promise<T> {
+    let transaction: EthereumTransaction | EthereumTransactionEIP1559;
     if (isOldStyleEthereumjsTx(tx)) {
       // legacy transaction from ethereumjs-tx package has no .toJSON() function,
       // so we need to convert to hex-strings manually manually
       transaction = {
-        to: this._normalize(tx.to),
-        value: this._normalize(tx.value),
-        data: this._normalize(tx.data),
+        to: this.#normalize(tx.to),
+        value: this.#normalize(tx.value),
+        data: this.#normalize(tx.data),
         chainId,
-        nonce: this._normalize(tx.nonce),
-        gasLimit: this._normalize(tx.gasLimit),
-        gasPrice: this._normalize(tx.gasPrice),
+        nonce: this.#normalize(tx.nonce),
+        gasLimit: this.#normalize(tx.gasLimit),
+        gasPrice: this.#normalize(tx.gasPrice),
       };
     } else {
       // new-style transaction from @ethereumjs/tx package
       // we can just copy tx.toJSON() for everything except chainId, which must be a number
       transaction = {
         ...tx.toJSON(),
         chainId,
-        to: this._normalize(tx.to),
-      };
+        to: this.#normalize(tx.to as unknown as Buffer),
```

Comment:
Hmm. I might not be following what's going on here, but it looks like `tx.to` here is supposed to be an `Address` type. I see that if you do pass an `Address` into `#normalize`, it will work because `normalize` calls `bufferToHex` which itself calls `ethUtil.toBuffer`, which tries in many ways to convert the given argument to a Buffer (`Address` has a `toBuffer` function that gets called). But it doesn't match the type signature we have here.

We could do something like this instead:
```suggestion
        to: this.#normalize(tx.to.toBuffer()),
```
Which _should_ work, except that `tx.to` can be `undefined` apparently. Maybe we could check if it's `undefined` first, then call `toBuffer`? :thinking:   That seems risky though, since this code is still being invoked from JavaScript, and who knows what gets passed in.

The safest option here might be to call `ethUtil.toBuffer` on this sooner. `toBuffer` is extremely flexible in what it accepts, and this parameter is already being passed through it so it would be functionally equivalent.

e.g.

```suggestion
        to: this.#normalize(ethUtil.toBuffer(tx.to)),
```